assembl/static2/js/app/components/administration/survey/themeForm.jsx

```diff
@@ -0,0 +1,186 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Translate, I18n } from 'react-redux-i18n';
+import { gql, withApollo } from 'react-apollo';
+import { Button, FormGroup, FormControl } from 'react-bootstrap';
+
+import { listThematicsToDelete, listPreviewsToDisplay } from '../../../actions/adminActions';
+import ImageUploader from '../../common/imageUploader';
+
+const GetThematics = gql`
+{
+  thematics(identifier:"survey") {
+    id,
+    titleEntries {
+      localeCode,
+      value
+    },
+    imgUrl,
+    video {
+      title,
+      description,
+      htmlCode
+    }
+    questions {
+      titleEntries {
+        localeCode,
+        value
+      }
+    }
+  }
+}
+`;
+
+export const updateTitle = (client, id, selectedLocale, titleEntryIndex, value) => {
+  let entryIndex = titleEntryIndex;
+  if (titleEntryIndex === -1) {
+    const res = client.readFragment({
+      id: `Thematic:${id}`,
+      fragment: gql`
+        fragment thematic on Thematic {
+          titleEntries
+        }
+      `
+    });
+    entryIndex = res.titleEntries.length;
+    const newTitleEntries = res.titleEntries;
+    newTitleEntries.push({
+      id: `Thematic:${id}.titleEntries.${entryIndex}`,
+      type: 'id'
+    });
+    client.writeFragment({
+      id: `Thematic:${id}`,
+      fragment: gql`
+        fragment thematic on Thematic {
+          titleEntries
+        }
+      `,
+      data: {
+        titleEntries: newTitleEntries,
+        __typename: 'Thematic'
+      }
+    });
+  }
+  client.writeFragment({
```

Comment:
the first `writeFragment` is when the lang entry already exists and the second one is to create a new one